design-proposals/data-import-cron-http-import.md

```diff
@@ -0,0 +1,78 @@
+
+# Overview
+This design provides an approach to allow DataImportCron to import HTTP sources.  
+The design will also demonstrate to the user how to operate with such import types.  
+
+## Motivation
+Currently, the DataImportCron allows the import of Registry Imports only. Recently, there was a demand to also allow HTTP import types.  
+The problem that arises from HTTP imports is that there is no convention between the different sources, so it's hard to know when the image is updated for each source in a generic way, which will make the polling process more difficult than standard registry sources.  
+One possible solution If the URL is static is to use a Get request with an If-Modified-Since header where we specify the date from which we want to check if there was a change. If there was a change since the specified date, the request will return with a status of 200OK, and then we know that the image has been updated since that date.  
+
+In the current situation we can enable HTTP imports by updating the DesiredDigest label.  
+
+## Goals
+
+* Allow the user to perform http imports manually with dataimportcron
+* Create a poller that will cover as many cases as possible for automatic updating
+
+## Non Goals
+
+* The poller will probably not cover all import cases and sometimes the user will have to manually update the digest
+
+## User Stories
+
+* As a user, I would like to import images from an HTTP source using DataImportCron
+* As a user, I would like the poller automatically update the image when it is needed
+* As a user, I would like to manually trigger an HTTP import with DataImportCron
+
+## Repos
+
+* **CDI**: Offers an API which serves as a mutable symbolic link to a Persistent Volume Claim (PVC). This link can be used in a DataVolume.
+* **CDI**: Provides an API called DataImportCron, designed to manage the periodic polling and importing of disk images as PVCs into a specific namespace (golden images by default). This functionality leverages the CDI import processes.
+
+## Implementation Phases
+
+**Phase1** - Manual Import Update
+* DataImportCron controller to handle HTTP imports
+* DataImportCron webhook/validation to support HTTP source type
+
+**Phase2** - Poller
+* DataImportCron poller to support HTTP source polling
+
+# Design
+
+If the URL is static:
```

Comment:
I don't understand what you mean by `If the URL is static` by definition the URL passed to the Datavolume or DataImportCron is static, it cannot be modified. Is there some other property we are interested in? I don't get it.

Reply:
Sometimes the image has the same URL and then we can use the way I mentioned, but sometimes the suffix changes when you upload a new version and we want to identify such a case, so we have to use another way.
When such case happens we want to update the `AnnSourceDesiredDigest` annotation to be the new suffix.

Reply:
Honestly, I don't think we should consider that particular scenario. From my perspective that is a completely different file. The URL that is passed in is always static and cannot change. There would be no mechanism that allows us to reliably know if a different file is newer than the original. I guess we could consider a 'template' url in combination with some version scheme, but that would be an enhancement IMO.